homeassistant/components/alarm_control_panel/mqtt.py

```diff
@@ -89,7 +99,7 @@ def message_received(topic, payload, qos):
             self._state = payload
             self.async_schedule_update_ha_state()
 
-        return mqtt.async_subscribe(
+        yield from mqtt.async_subscribe(
```

Comment:
Was this a bug?

Reply:
No, but it is required now as multiple topics can be subscribed (status and availability).

Reply:
Ok, for clarity I think we should add the `@asyncio.coroutine` annotation to all `async_added_to_hass` methods in the entity classes affected by this PR. I think that should have been done previously, so if you can do that it's :+1:.

Reply:
Also update the docstrings for these methods. Ie they don't return a coroutine object anymore.

Reply:
My last comment came out weird, of course the coroutine function will still return a coroutine object but it doesn't explicitly return something, so I think we can remove the last sentence in the docstring.